specs/keys/keystore.md

```diff
@@ -0,0 +1,67 @@
+# Keystores
+
+A keystore is a JSON file which store an encrypted version of a user's private key. It is designed to be an easy-to-implement format for storing and exchanging keys. Furthermore, this specification is designed to utilize as few crypto-constructions and make a minimal number of security assumptions
+
+![Keystore Diagram](./keystore.png)
+
+## Definition
+
+Private key is obtained by taking the bitwise XOR of the `ciphertext` and the `derived_key`. The `derived_key` is obtained by running scrypt with the user-provided password and the `scryptparams` obtained from within the keystore file as parameters. If a keystore file is being generated for the first time, the `salt` KDF parameter must be obtained from a CSPRNG. The `ciphertext` is simply read from the keystore file. The length of the `ciphertext` and the output key length of scrypt.
+
+```python
+def decrypt_keystore(password, dklen, n, p, r, salt, ciphertext) -> bytes
+    assert len(ciphertext) == dklen
+    derived_key = scrypt(password, dklen, n, p, r, salt)
+    return bytes(a ^ b for a, b in zip(derived_key, ciphertext))
+```
+
+## UUIDs
+
+The ID provided in the keystore is randomly generated and is designed to be used as a 128-bit proxy for referring to a set of keys or account. This level of abstraction provides a means of preserving privacy for a secret-key.
+
+## Test vectors
+
+**The following is not a valid test. It is a placeholder that will be populated with valid data soon** (tm):
+
+Test values:
+
+* Password: `testpassword`
+* Secret: `7a28b5ba57c53603b0b07b56bba752f7784bf506fa95edc395f5cf6c7514fe9d`
+
+```json
```

Comment:
Yeah, absolutely! This is one of those cases where deviating from YAML for a test seems like the sensible option. I'm not necessarily choosing JSON because of its use in Eth1, but more that it seems to be the sensible choice due to its ubiquity. I'm open to alternative suggestions, but will run with JSON for the tests unless others have strong opinions.